sklearn/manifold/t_sne.py

```diff
@@ -402,19 +632,50 @@ def __init__(self, n_components=2, perplexity=30.0,
         self.metric = metric
         self.init = init
         self.verbose = verbose
-        self.random_state = random_state
+        self.random_state = check_random_state(random_state)
```

Comment:
This is the reason of the failure. This should be:

```python
self.random_state = random_state
```

as the `__init__` method should mutate the hyperparameters.

And in `fit` turn it into an RNG in a local variable:

```python
random_state = check_random_state(random_state)
```